compiler/rustc_index/src/lib.rs

```diff
@@ -1,17 +1,20 @@
 #![deny(rustc::untranslatable_diagnostic)]
 #![deny(rustc::diagnostic_outside_of_impl)]
-#![feature(allow_internal_unstable)]
-#![feature(extend_one)]
-#![feature(min_specialization)]
-#![feature(new_uninit)]
-#![feature(step_trait)]
-#![feature(stmt_expr_attributes)]
-#![feature(test)]
+#![cfg_attr(feature = "nightly", feature(allow_internal_unstable))]
+#![cfg_attr(feature = "nightly", feature(extend_one))]
+#![cfg_attr(feature = "nightly", feature(min_specialization))]
+#![cfg_attr(feature = "nightly", feature(new_uninit))]
+#![cfg_attr(feature = "nightly", feature(step_trait))]
+#![cfg_attr(feature = "nightly", feature(stmt_expr_attributes))]
+#![cfg_attr(feature = "nightly", feature(test))]
```

Comment:
> It may make sense to combine the features to a single list here - feature(allow_internal_unstable, extend_one, ...).

I combined them.

> Did you try building this crate with only feature = "rustc_serialize" or only feature = "rustc_macros"?
Does it require enabling any nightly features in this case?

It builds without nightly features and just with a nightly compiler.